### PR TITLE
feat: KPI-148 create endpoint to upload file to S3

### PR DESCRIPTION
**Pull request by dsgarcia8**
## Jira Link

[Go to Jira](https://kpinetwork.atlassian.net/browse/KPI-148)

## Types of changes

- [ ] Hotfix (Bug detected)
- [x] Feature (Issue from an Active Sprint)

## Checklist:

- [x] Unit Test.
- [x] Coverage.

### CODE REVIEW

File: infrastructure/buckets/outputs.tf [hunk 1]
```diff
@@ -104,10 +104,16 @@ output "object_references" {
...
      }
 
-     "get_all_public_companies_function_bucket" : {
+    "get_all_public_companies_function_bucket" : {
       etag : aws_s3_bucket_object.get_all_public_companies_function_object.etag,
       key : aws_s3_bucket_object.get_all_public_companies_function_object.key,
       bucket : aws_s3_bucket_object.get_all_public_companies_function_object.bucket
     }
+
+    "upload_data_s3_trigger_function_bucket" : {
```
Comment (kellycastrof) [inline infrastructure/buckets/outputs.tf:113]: Can we change the name?
The lambda is just a connection with API Gateway for and endpoint call, "trigger" can be confused name for the actual functionality.
The lambda that starts running the job in glue is a trigger.

**Review by kellycastrof — CHANGES_REQUESTED** [review]

File: infrastructure/buckets/outputs.tf (hunk 1 above)
Comment (dsgarcia8) [reply infrastructure/buckets/outputs.tf]: resources were changed to 'upload_file_s3' and endpoint is now 'upload_file'

File: src/handlers/upload_data/upload_data_s3_trigger_handler.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+def upload_file(bucket_name: str, file_info: dict) -> None:
+    filename = file_info.get("fileName")
+    file = file_info.get("file")
+
+    if not file or not filename:
+        raise AppError("No file provided")
+    if not filename.endswith((".csv")):
+        raise AppError("Extension not allowed")
+
+    boto3.Session(
```
[the remark this answers is not in the document]
Comment (dsgarcia8) [reply src/handlers/upload_data/upload_data_s3_trigger_handler.py]: New function added: 'get_s3_client'

**Review by FreddyJR1995 — APPROVED** [review]
Looks good to me! Great job!!

**Pull request merged**